rust_src/src/coding.rs

```diff
@@ -98,4 +103,38 @@ pub fn encode_file_name(fname: LispStringRef) -> LispStringRef {
     unsafe { c_encode_file_name(fname.into()) }.into()
 }
 
+/// Read a coding system from the minibuffer, prompting with string PROMPT.
+/// If the user enters null input, return second argument DEFAULT-CODING-SYSTEM.
+/// Ignores case when completing coding systems (all Emacs coding systems
+/// are lower-case).
+#[lisp_fn(min = "1")]
+pub fn read_coding_system(prompt: LispObject, mut default_coding_system: LispObject) -> LispObject {
+    let count = c_specpdl_index();
+
+    if let Some(s) = default_coding_system.as_symbol() {
+        default_coding_system = s.symbol_name();
+    }
+
+    unsafe {
+        specbind(Qcompletion_ignore_case, Qt);
+    }
+    let val = completing_read(
+        prompt,
+        unsafe { globals.Vcoding_system_alist },
+        Qnil,
+        Qt,
+        Qnil,
+        Qcoding_system_history,
+        default_coding_system,
+        Qnil,
+    );
+    unbind_to(count, Qnil);
+
+    if LispStringRef::from(val).size() == 0 {
```

Comment:
Since both arms need a LispStringRef, perhaps it should saved in a temp variable?

```
let tem: LispStringRef = val.into();
if tem.is_empty() {
    Qnil
} else {
    intern_lisp(tem, None)
}
```